### PR TITLE
Feature/refactroring

### PR DESCRIPTION
**Pull request by diamandv**
if you see it - i did it!) Would be more!

### CODE REVIEW

File: src/main/java/Starter.java
Comment (BogdanParashchak) [inline src/main/java/Starter.java:1]: to be deleted
we dont't need it any more bacause of JUnit tests

File: src/test/java/SomeTwentyMethodsTest.java [hunk 1]
```diff
@@ -92,68 +98,170 @@ void givenEmptyIntArray_whenConvertToCharArray_thenReturnEmptyCharArray() {
...
 
+    //Task 4.1
     @Test
-    void givenThreeEqualInts_whenCompareInts_thenReturnLargestOfThreeInts() {
+    void givenThreeEqualInts_whenFindTheLargestOfTwoInts_thenReturnLargestOfThreeInts() {
         //given
         int a = 5;
         int b = 5;
         int c = 5;
+        assertEquals(a,b,c);
```
Comment (BogdanParashchak) [inline src/test/java/SomeTwentyMethodsTest.java:228]: formatting

File: src/test/java/SomeTwentyMethodsTest.java [hunk 2]
```diff
@@ -92,68 +98,170 @@ void givenEmptyIntArray_whenConvertToCharArray_thenReturnEmptyCharArray() {
...
+    }
+
+    //Task 4.2
+    @Test
+    void givenThreeEqualInts_whenFindTheLargestOfThreeIntsUsingTernaryOperator_thenReturnLargestOfThreeInts() {
+        //given
+        int a = 7;
+        int b = 7;
+        int c = 7;
+        assertEquals(a,b,c);
```
Comment (BogdanParashchak) [inline src/test/java/SomeTwentyMethodsTest.java:255]: formatting

File: src/test/java/SomeTwentyMethodsTest.java [hunk 3]
```diff
@@ -255,37 +363,36 @@ void givenNullIntArray_whenFindIndexOfIntegersThatMatchesWithInt_thenReturnIndex
...
+    void givenIntAndIntArrayWithMatchingValue_whenFindValueOfIndexInIntArrayThatMatchesWithInt_thenReturnFirstIndexWithMatchingValueButNotTheOtherNext() {
         //given
         int[] integers = new int[]{Integer.MIN_VALUE, Integer.MAX_VALUE, 6, 6, 6};
         int intToFind = 6;
         //when
-        int actualIndex = someTwentyMethods.findIndexOfIntegersThatMatchesWithValue(integers, intToFind);
+        int actualIndex = someTwentyMethods.findIndexOfIntegersThatMatchesWithItsValue(integers, intToFind);
         //then
-        assertNotEquals(3 | 4, actualIndex);
+        assertEquals(2,actualIndex);
```
Comment (BogdanParashchak) [inline src/test/java/SomeTwentyMethodsTest.java:384]: formatting

File: src/test/java/SomeTwentyMethodsTest.java [hunk 4]
```diff
@@ -420,222 +524,327 @@ void givenIntWithZeroIntValue_whenFindFactorialOfGivenInt_thenReturnOne() {
...
     //Test 12
     @Test
     void givenIntAndIntArray_whenFindMultipleNumbersOfIntInArray_thenReturnArrayListOfFoundMultipleNumbers() {
+        //given
         int[] integersArray = new int[]{Integer.MIN_VALUE, 2, 3, 4, 5, 6, 7, 8, 9, Integer.MAX_VALUE};
         int integer = 4;
         ArrayList<Integer> expectedArray = new ArrayList<>();  //this part of code is not beautiful but work
         expectedArray.add(-2147483648);
         expectedArray.add(4);
         expectedArray.add(8);
```
Comment (BogdanParashchak) [inline src/test/java/SomeTwentyMethodsTest.java:584]: List<Integer> expectedArray = List.of(......)

File: src/test/java/SomeTwentyMethodsTest.java [hunk 5]
```diff
@@ -420,222 +524,327 @@ void givenIntWithZeroIntValue_whenFindFactorialOfGivenInt_thenReturnOne() {
...
+        //when
         ArrayList<Integer> actualArray = someTwentyMethods.findMultipleNumbersOfArray(integersArray, integer);
```
Comment (BogdanParashchak) [inline src/test/java/SomeTwentyMethodsTest.java:586]: List Integer actualArray = someTwentyMethods.findMultipleNumbersOfArray(integersArray, integer);

File: src/test/java/SomeTwentyMethodsTest.java [hunk 6]
```diff
@@ -420,222 +524,327 @@ void givenIntWithZeroIntValue_whenFindFactorialOfGivenInt_thenReturnOne() {
...
     void givenIntAndEmptyIntArray_whenFindMultipleNumbersOfIntInArray_thenReturnEmptyArrayList() {
+        //given
         int[] integersArray = new int[]{};
         int integer = 3;
         ArrayList<Integer> expectedArray = new ArrayList<>();
+        //when
         ArrayList<Integer> actualArray = someTwentyMethods.findMultipleNumbersOfArray(integersArray, integer);
+        //then
         assertEquals(expectedArray, actualArray);
     }
```
Comment (BogdanParashchak) [inline src/test/java/SomeTwentyMethodsTest.java:649]: all the occurences of
ArrayList<Integer> expectedArray =
and
ArrayList<Integer> actualArray =
replace with
List<Integer> expectedArray = 
and
List<Integer> actualArray =

File: src/test/java/SomeTwentyMethodsTest.java [hunk 7]
```diff
@@ -420,222 +524,327 @@ void givenIntWithZeroIntValue_whenFindFactorialOfGivenInt_thenReturnOne() {
...
 
+    //Test 16
+    @Test
+    void givenTwoEqualIntArray_whenCompareElementsOfTwoArrays_thenReturnArrayOfElementsThatDontMatch() {
+        //given
+        int[] intArray1 = new int[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+        int[] intArray2 = new int[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+        ArrayList<Integer> expectedIntegers = new ArrayList<>();
+        //when
+        ArrayList<Integer> actualIntegers = someTwentyMethods.getArrayOfNonMatchingIndexes(intArray1, intArray2);
```
Comment (BogdanParashchak) [inline src/test/java/SomeTwentyMethodsTest.java:830]: List actual......

File: src/test/java/SomeTwentyMethodsTest.java [hunk 8]
```diff
@@ -420,222 +524,327 @@ void givenIntWithZeroIntValue_whenFindFactorialOfGivenInt_thenReturnOne() {
...
+    void givenTwoIntArrayWithDifferentValues_whenCompareElementsOfTwoArrays_thenReturnArrayOfElementsThatDontMatch() {
+        //given
+        int[] intArray1 = new int[]{1, 5, 3, 7, 5, 6, 2, 8, 9, 10};
+        int[] intArray2 = new int[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+        ArrayList<Integer> expectedIntegers = new ArrayList<>();
+        expectedIntegers.add(5);
+        expectedIntegers.add(7);
+        expectedIntegers.add(2);
+        //when
+        ArrayList<Integer> actualIntegers = someTwentyMethods.getArrayOfNonMatchingIndexes(intArray1, intArray2);
```
Comment (BogdanParashchak) [inline src/test/java/SomeTwentyMethodsTest.java:845]: List = .....

**Review by BogdanParashchak — CHANGES_REQUESTED** [review]

File: src/main/java/Starter.java
Comment (diamandv) [reply src/main/java/Starter.java:1]: done

File: src/test/java/SomeTwentyMethodsTest.java (hunk 1 above)
Comment (diamandv) [reply src/test/java/SomeTwentyMethodsTest.java]: done

File: src/test/java/SomeTwentyMethodsTest.java (hunk 2 above)
Comment (diamandv) [reply src/test/java/SomeTwentyMethodsTest.java]: done

File: src/test/java/SomeTwentyMethodsTest.java (hunk 3 above)
Comment (diamandv) [reply src/test/java/SomeTwentyMethodsTest.java]: done

File: src/test/java/SomeTwentyMethodsTest.java (hunk 4 above)
Comment (BogdanParashchak) [reply src/test/java/SomeTwentyMethodsTest.java:684]: in futher commits

File: src/test/java/SomeTwentyMethodsTest.java (hunk 5 above)
Comment (BogdanParashchak) [reply src/test/java/SomeTwentyMethodsTest.java:686]: in futher commits please

File: src/test/java/SomeTwentyMethodsTest.java (hunk 7 above)
Comment (BogdanParashchak) [reply src/test/java/SomeTwentyMethodsTest.java:930]: in futher commits please

File: src/test/java/SomeTwentyMethodsTest.java (hunk 8 above)
Comment (BogdanParashchak) [reply src/test/java/SomeTwentyMethodsTest.java:945]: in futher commits please

File: src/test/java/SomeTwentyMethodsTest.java (hunk 6 above)
Comment (BogdanParashchak) [reply src/test/java/SomeTwentyMethodsTest.java:749]: in futher commits please

**Review by BogdanParashchak — APPROVED** [review]

**Pull request merged**